caddy.go

```diff
@@ -606,16 +606,15 @@ func executeDirectives(inst *Instance, filename string,
 	return nil
 }
 
-func startServers(serverList []Server, inst *Instance, restartFds map[string]restartTriple) error {
+func startServers(serverList []Server, inst *Instance, restartFds map[string]restartTriple) (err error) {
 	errChan := make(chan error, len(serverList))
+	var (
+		ln  net.Listener
+		pc  net.PacketConn
+	)
 
 	for _, s := range serverList {
-		var (
```

Comment:
I don't think we could move these out of the loop, because we check `ln` and `pc` below (line 651 and 657).